File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <unordered_map>
using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    unordered_map<string,int> m;
    for(auto x: participant){
        m[x]++;
    }
    for(auto x:completion){
        m[x]--;
    }
    vector<pair<string,int>> v(m.begin(),m.end());
    for(auto x:v){
        if(x.second==1) answer=x.first;
    }
    return answer;
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;
class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        Map<String,Integer> map = new HashMap<>();
        
        for(String x : participant) {
            map.put(x,map. getOrDefault(x,0)+1);
        }
        for(String y : completion) {
            map.put(y, map.get(y)-1);
        }
        
        for(Map.Entry<String,Integer> entry : map.entrySet()) {
            if(entry.getValue()==1) {
                answer = entry.getKey();
            }
        }
        return answer;
    }
}